node/network/protocol/src/peer_set.rs

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! All peersets and protocols used for parachains.
+
+use sc_network::config::{NonDefaultSetConfig, SetConfig};
+use std::borrow::Cow;
+use strum::{IntoEnumIterator, EnumIter};
+
+/// The peer-sets and thus the protocols which are used for the network.
+#[derive(Debug, Clone, Copy, PartialEq, EnumIter)]
+pub enum PeerSet {
+	/// The validation peer-set is responsible for all messages related to candidate validation and communication among validators.
+	Validation,
+	/// The collation peer-set is used for validator<>collator communication.
+	Collation,
+	/// Direct peer connections for availability distribution (and recovery?).
+    AvailabilityDistribution,
+}
+
+/// Protocol name as understood in substrate.
+///
+/// Ideally this would be defined in substrate as a newtype.
```

Comment:
Making the newtype would be a trivial but a pervasive change in substrate, likely not worth it, as being in substrate, it would also break code for other users. Type safety gains are probably low, as we are not using Cow<'static,str> much. 

Just putting this out there to see what people think.